cli/commands/pack.go

```diff
@@ -62,17 +62,17 @@ func addTarantoolDepIfNeeded(ctx *context.Ctx) error {
 		return nil
 	}
 
-	var tarantoolVersion string
-	if ctx.Pack.Type == pack.RpmType {
-		tarantoolVersion = strings.SplitN(ctx.Tarantool.TarantoolVersion, "-", 2)[0]
-	} else if ctx.Pack.Type == pack.DebType {
-		tarantoolVersion = ctx.Tarantool.TarantoolVersion
+	tarantoolMinVersion, err := pack.GetTarantoolMinVersion(ctx.Tarantool.TarantoolVersion, ctx.Pack.Type)
+	if err != nil {
+		return fmt.Errorf("Failed to get minimum Tarantool version: %s", err)
 	}
 
-	if err := ctx.Pack.Deps.AddTarantool(tarantoolVersion); err != nil {
-		return fmt.Errorf("Failed to get tarantool dependency: %s", err)
+	tarantoolMaxVersion, err := common.GetNextMajorVersion(ctx.Tarantool.TarantoolVersion)
+	if err != nil {
+		return fmt.Errorf("Failed to get maximum Tarantool version: %s", err)
 	}
 
+	ctx.Pack.Deps.AddTarantool(tarantoolMinVersion, tarantoolMaxVersion)
```

Comment:
I do not quite understand what exactly you are talking about. Can you explain please?

Reply:
AFAIU, Leonid means that before the patch `AddTarantool()` accepts a current tarantool version (as opposite to a border / borders), while after the patch it accepts borders.

Reply:
In other words, `GetNextMajorVersion()` goes from the `AddTarantool()` helper to this function.

Reply:
Yep, why can't we leave the logic with `GetNextMajorVersion` in the `AddTarantool` function.

Reply:
We can move the call `GetNextMajorVersion` to `AddTarantool`, but:

We can't call `pack.GetTarantoolMinVersion` in `AddTarantool()` function (`common` package will import `pack` package -> cycle imports), also we can't describe `GetTarantoolMinVersion` in `common` package, because we need constant [`pack.RpmType`](https://github.com/tarantool/cartridge-cli/blob/fdd349b4f99cdbb4f3f6b91300d97183ff1859b3/cli/pack/common.go#L62). We can't move `GetTarantoolMinVersion` to `AddTarantool()`, and in this case, we anyway have to pass two arguments `tarantoolMinVersion` and  `ctx.Tarantool.TarantoolVersion` (bcs `GetNextMajorVersion` will work only with him - if we pass tarantoolMinVersion, we will get error). And this looks ugly.